vllm/entrypoints/openai/api_server.py

```diff
@@ -669,6 +670,17 @@ async def fake_stream_generator() -> AsyncGenerator[str, None]:
                         help="The model name used in the API. If not "
                         "specified, the model name will be the same as "
                         "the huggingface name.")
+    parser.add_argument("--chat-template",
+                        type=str,
+                        default=None,
+                        help="The file path to the chat template, "
+                        "or the template in single-line form "
+                        "for the specified model")
+    parser.add_argument("--response-role",
```

Comment:
I think this largely depends on the templates itself right? By default I don't think this is needed.

Reply:
The response-role? Some models may not use user/assistant as the role names. The templates are free with that. I defaulted to the same behavior as the OpenAI API.

Reply:
Ye but we already have that in the message['role'] right? not sure why we need this

Reply:
Yes it usually alternates between user/assistant...but given the following request, how would we know the role to respond with?
```
curl http://0.0.0.0:8000/v1/chat/completions \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer " \
  -d '{
    "model": "Tostino/Inkbot-13B-8k-0.2",
    "stream": false,
    "add_generation_prompt": true,
    "echo": true,
    "n": 1,
    "messages": [
	{"role": "meta-current_date", "content": "2023-10-20"},
	{"role": "meta-task_name", "content": "general"},
	{"role": "system", "content": "You are a helpful assistant."},
	{"role": "user", "content": "What is the capital of the USA?"}	
	]
  }'
  ```
  
Previously, it was hard-coded as `assistant`. Now it is configurable with that cli argument. Ideally, this would be additional metadata about how to use the template that would be stored somewhere in the tokenizer...but HF didn't think that far ahead.

Reply:
Users should be responsible for doing few-shot prompt right?

Reply:
No, that is totally unnecessary for a lot of models and just eats up context space.